### PR TITLE
Ship movement

### PR DESCRIPTION
**Pull request by ThierryBerger**
I'm familiarizing with the codebase by implementing random features, let me know if anything is useful to you:

### Done so far

- player movement independent from FPS by using time resource
- small rotation of the player ship if going up or down ; a side effect is that we now shoot up or down, I quite liked it as it makes the game a bit easier, but eventually we could force a shoot straight..

### Ideas for other improvements;
- we could share movement code between ai and player by reusing WantedMovement
- fire reactor particles when going right ? another particle effect when going left ?
- ships are cut in half, I'm not sure why, but probably the camera far plane is cutting them, or there is a background hiding them 🤔 
- explosion of the ship, with a freeze à la https://www.youtube.com/watch?v=OdVkEOzdCPw&t=3s

### Current state as a gif:

https://user-images.githubusercontent.com/2290685/200613641-bf97fd12-bb41-4c8a-a914-e1ffc62f9573.mp4

### CODE REVIEW

File: src/game/player.rs [hunk 1]
```diff
@@ -62,31 +72,73 @@ pub fn spawn_player(
...
+            keyboard_input.pressed(KeyCode::Down),
+        );
+        if up ^ down {
+            if up {
+                wanted_direction.0.y = 1f32;
+            } else {
+                wanted_direction.0.y = -1f32;
+            }
+        } else {
+            wanted_direction.0.y = 0f32;
```
Comment (ThierryBerger) [inline src/game/player.rs:89]: might  be cleaner as a match

File: src/game/vfx/mod.rs [hunk 2]
```diff
@@ -60,7 +60,7 @@ fn shake_camera(
             t.translation = Vec3::new(
                 magnitude_at_time * theta.cos(),
                 magnitude_at_time * theta.sin(),
-                CAMERA_FAR,
+                CAMERA_FAR - 100f32,
```
Comment (ThierryBerger) [inline src/game/vfx/mod.rs:63]: I tried to fix the models cut in half but I don't think that fixed it

**Comment by LarsDu** [comment]
Kudos for playing around with this codebase.

I'm inclined to close this as I'm not a fan of this from a gameplay perspective.  I mainly want to keep movement simple as with classic SHMUPS like U.N. Squadron

**Comment by LarsDu** [comment]
Sidenote: This codebase actually needs some linting and cleanup. If I have time to get back to this, expect some signicant changes after bevy 0.9 lands

**Comment by ThierryBerger** [comment]
Thanks for the feedback and the reference ; I admit I went overboard with the ability to shoot up and down 😅 

I mainly wanted to bring to your attention that "simple" polish for the ship animation can make the game more alive, if we take your reference UN squadron (if I researched that correctly): The ship has a different stance when going up or down:
ship going down
_ship going down_

ship going up
_ship going up_

If you're fine with that I can focus on not impacting gameplay while adding some polish enhancement :) _(or find me on discord if you want to discuss anything else)_

**Pull request closed without merging**